src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { FC, useState, useCallback, useMemo } from 'react';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+
+import { peopleFromServer } from '../../data/people';
+import { Person } from '../../types/Person';
+import { filterPeople } from '../../utils/filterPeople';
+import { DropdownMenu } from '../DropdownMenu/DropdownMenu';
+
+type Props = {
+  selectedPerson: Person | null;
+  onSelect: React.Dispatch<React.SetStateAction<Person | null>>;
+  delay?: number;
+};
+
+export const Dropdown: FC<Props> = ({
+  selectedPerson,
+  onSelect,
+  delay = 300,
+}) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isActive, setIsActive] = useState(false);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), []);
+
+  const filteredPeople = useMemo(
+    () => filterPeople(peopleFromServer, { appliedQuery }),
+    [appliedQuery],
+  );
+
+  const visiblePeople = selectedPerson ? peopleFromServer : filteredPeople;
+
+  const handleInputFocus = () => {
+    setIsActive(true);
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value.trimStart());
+    applyQuery(event.target.value);
+    onSelect(null);
+  };
```

Comment:
The `handleInputChange` function is not following the recommended naming conventions for event handlers in React. According to the [naming conventions](https://medium.com/javascript-in-plain-english/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364), the function should be named `onInputChange`.